infra/kops/tf_output_to_kops_values.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+import os
+import json
+import subprocess
+import argparse
+
+
+CWD = os.path.abspath(os.path.dirname(__file__))
+TF_BASE_DIR = os.path.abspath(os.path.join(
+    CWD, '../terraform'
+))
+
+
+def flatten_tf_output(d):
+    """Convert terraform output JSON to a flat dict
+
+    >>> flatten_tf_output({
+    ...    'vpc_cidr': {
+    ...        'sensitive': False,
+    ...        'type': 'string',
+    ...        'value': '192.168.0.0/16'
+    ...    }
+    ... })
+    {'vpc_cidr': '192.168.0.0/16'}
+    """
+    return {k: v['value'] for k, v in d.items()}
+
+
+def subnet_attr_lists_to_dicts(subnets):
+    """Transpose Terraform dict of lists to a list of dicts. e.g:
+
+    >>> subnet_attr_lists_to_dicts({
+    ...     'availabilityZones': [
+    ...         'eu-west-1a',
+    ...         'eu-west-1b'
+    ...     ],
+    ...     'cidrs': [
+    ...         '192.168.10.0/24',
+    ...         '192.168.14.0/24'
+    ...     ],
+    ...     'ids': [
+    ...         'subnet-7d72600b',
+    ...         'subnet-7c376924'
+    ...     ]
+    ... })
+    [{'availabilityZone': 'eu-west-1a', 'cidr': '192.168.10.0/24', 'id': \
+'subnet-7d72600b'}, {'availabilityZone': 'eu-west-1b', 'cidr': \
+'192.168.14.0/24', 'id': 'subnet-7c376924'}]
+    """
+    return [
+        {'availabilityZone': az, 'cidr': cidr, 'id': sid}
+        for az, cidr, sid in zip(subnets['availabilityZones'],
+                                 subnets['cidrs'],
+                                 subnets['ids'])
+    ]
+
+
+def get_tf_resource_dir(tf_resources):
+    return os.path.abspath(os.path.join(TF_BASE_DIR, tf_resources))
+
+
+def get_tf_output(tf_resources):
+    os.chdir(get_tf_resource_dir(tf_resources))
+    return flatten_tf_output(
+        json.loads(
+            subprocess.getoutput("terraform output -json")))
```

Comment:
it would be useful to also bubble up error terraform messages:

-    return flatten_tf_output(
-        json.loads(
-            subprocess.getoutput("terraform output -json")))
+    exitcode, output = subprocess.getstatusoutput("terraform output -json")
+    assert exitcode == 0, output
+    return flatten_tf_output(json.loads(output))